### PR TITLE
Add Pasqal support

### PR DESCRIPTION
**Pull request by tncc**
Example code:

from azure.quantum import Workspace
from azure.quantum.target.rigetti import Result, Rigetti, RigettiTarget, InputParams
from azure.quantum.target.pasqal import Result, Pasqal, PasqalTarget, InputParams
import cirq

workspace = Workspace(
    resource_id="WORKSPACE_RESOURCE_ID_HERE",
    location="LOCATION_HERE"
)

target = Pasqal(
    workspace=workspace,
    name=PasqalTarget.SIM_EMU_FREE
)

with open("pasqal.sim", "r") as f:
    test_pulser = f.read()

job = target.submit(
    input_data=test_pulser, 
    name="test_pulser", 
)

print(f"Queued job: {job.id}")
job.wait_until_completed()

print(f"Job completed with state: {job.details.status}")
result = Result(job)  # This throws an exception if the job failed

print(result.data)

### CODE REVIEW

File: azure-quantum/azure/quantum/target/pasqal/target.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        """Returns a list of QPU targets"""
+        return [
+            PasqalTarget.QPU.value,
+            PasqalTarget.QPU_FRESNEL.value
+        ]
+
+    def num_qubits(target_name) -> int:
+        """Returns the number of qubits supported by the given target"""
+        if target_name == PasqalTarget.SIM.value:
+            return 20 # TODO temp value, need info from Pasqal
```
[the remark this answers is not in the document]
Comment (tncc) [reply azure-quantum/azure/quantum/target/pasqal/target.py]: File updated, TODO is now todone

**Comment by shenyingjun** [comment]
Could you add a code example to show how user will use qdk to submit job to pasqal in the PR description?

File: azure-quantum/azure/quantum/target/pasqal/result.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+        if job.details.status != "Succeeded":
+            raise RuntimeError(
+                "Cannot retrieve results as job execution failed "
+                f"(status: {job.details.status}."
+                f"error: {job.details.error_data})"
+            )
+        self.data = cast(Dict[str, int], json.loads(job.download_data(job.details.output_data_uri)))
+
+    def __getitem__(self, register_name: str) -> int:
```
Comment (shenyingjun) [inline azure-quantum/azure/quantum/target/pasqal/result.py:43]: What's the use case of this method?

**Comment by shenyingjun** [comment]
@masenol could you review the PR and confirm parameters are all set correctly

File: azure-quantum/azure/quantum/target/pasqal/result.py (hunk 2 above)
Comment (tncc) [reply azure-quantum/azure/quantum/target/pasqal/result.py:43]: Not very useful but boilerplate as used in other targets

Comment (shenyingjun) [reply azure-quantum/azure/quantum/target/pasqal/result.py:43]: @masenol could you also confirm the output from pasqal is a string-int dictionary?

Comment (masenol) [reply azure-quantum/azure/quantum/target/pasqal/result.py:43]: @shenyingjun @tncc confirmed the output format to be a string-int dictionary from Pasqal.

File: azure-quantum/tests/unit/recordings/test_job_submit_pasqal_default_input_params.yaml [hunk 3]
```diff
@@ -0,0 +1,769 @@
...
+    headers:
+      accept-ranges:
+      - bytes
+      content-length:
+      - '383'
+      content-range:
+      - bytes 0-352/353
+      content-type:
+      - application/json
+      x-ms-blob-content-md5:
```
[the remark this answers is not in the document]
Comment (masenol) [reply azure-quantum/tests/unit/recordings/test_job_submit_pasqal_default_input_params.yaml:752]: @shenyingjun Anything we should do for these devskim alerts?

**Comment by masenol** [comment]
> @masenol could you review the PR and confirm parameters are all set correctly

File: azure-quantum/tests/unit/test_pasqal.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
...59172, \\"min_retarget_interval\\": null, \\"fixed_retarget_t\\": null, \\"max_targets\\": null, \\"clock_period\\": 4, \\"min_duration\\": 16, \\"max_duration\\": 100000000, \\"mod_bandwidth\\": 2, \\"eom_config\\": {\\"limiting_beam\\": \\"RED\\", \\"max_limiting_amp\\": 251.32741228718345, \\"intermediate_detuning\\": 4398.22971502571, \\"controlled_beams\\": [\\"BLUE\\", \\"RED\\"], \\"mod_bandwidth\\": 11}}], \\"name\\": \\"Fresnel\\", \\"dimensions\\": 2, \\"rydberg_level\\": 60, \\"min_atom_distance\\": 5, \\"max_atom_num\\": 20, \\"max_radial_distance\\": 35, \\"interaction_coeff_xy\\": null, \\"supports_slm_mask\\": false, \\"max_layout_filling\\": 0.5, \\"reusable_channels\\": false, \\"pre_calibrated_layouts\\": [], \\"is_virtual\\": false}, \\"layout\\": {\\"coordinates\\": [[-12.5, 4.330127], [-10.0, 0.0], [-7.5, -4.330127], [-7.5, 4.330127], [-5.0, -8.660254], [-5.0, 0.0], [-5.0, 8.660254], [-2.5, -4.330127], [-2.5, 4.330127], [0.0, -8.660254], [0.0, 0.0], [0.0, 8.660254], [2.5, -4.330127], [2.5, 4.330127], [5.0, -8.660254], [5.0, 0.0], [5.0, 8.660254], [7.5, -4.330127], [7.5, 4.330127], [10.0, 0.0]], \\"slug\\": \\"TriangularLatticeLayout(20, 5.0\\\\u00b5m)\\"}}"}"""
```
Comment (masenol) [inline azure-quantum/tests/unit/test_pasqal.py:16]: It'd be good to capture a sample with the pulser library generating these json inputs.

**Comment by masenol** [comment]
> @masenol could you review the PR and confirm parameters are all set correctly

Looks good to me @shenyingjun @tncc

File: azure-quantum/tests/unit/test_pasqal.py (hunk 4 above)
Comment (tncc) [reply azure-quantum/tests/unit/test_pasqal.py:16]: Reserving this for future work.

**Pull request merged**